translate-c: RHS of pointer arithmetic compound assignment coerced to wrong type

### Zig Version

0.14.0-dev.23+d9bd34fd0

### Steps to Reproduce and Observed Behavior

Use `zig translate-c` to translate certain expressions of the form `pointer += unsigned`, for example:
```c
void pointer_add(void) {
    char *str;
    unsigned int uint = 1;

    str += uint;
}
```

The result is:
```zig
pub export fn pointer_add() void {
    var str: [*c]u8 = undefined;
    _ = &str;
    var uint: c_uint = 1;
    _ = &uint;
    str += @as([*c]u8, @ptrFromInt(uint));
}
```

which doesn't make sense (adding a pointer to a pointer), and the compiler agrees:
```
$ zig translate-c pa.c > pa.zig
$ zig build-obj pa.zig
pa.zig:62:12: error: expected type 'usize', found '[*c]u8'
    str += @as([*c]u8, @ptrFromInt(uint));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This happens specifically for compound assignment to a pointer with `+=` or `-=`, when the type of the right-hand side is `unsigned int`, `unsigned long`, or `unsigned long long`.  The following all translate correctly:
```c
ptr = ptr + uint;
ptr += signed_int;
ptr += ushort;
ptr += uchar;
```

### Expected Behavior

The translated version should read `ptr += add;` or something else sensible.

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -3824,8 +3824,9 @@ fn transCreateCompoundAssign(
     const lhs_qt = getExprQualType(c, lhs);
     const rhs_qt = getExprQualType(c, rhs);
     const is_signed = cIsSignedInteger(lhs_qt);
+    const is_ptr_arithmetic = qualTypeIsPtr(lhs_qt) and cIsInteger(rhs_qt);
     const is_ptr_op_signed = qualTypeIsPtr(lhs_qt) and cIsSignedInteger(rhs_qt);
-    const requires_cast = !lhs_qt.eq(rhs_qt) and !is_ptr_op_signed;
+    const requires_cast = !lhs_qt.eq(rhs_qt) and !is_ptr_arithmetic;
 
     if (used == .unused) {
         // common case

--- a/test/cases/run_translated_c/compound_assignments_with_pointer_arithmetic.c
+++ b/test/cases/run_translated_c/compound_assignments_with_pointer_arithmetic.c
@@ -1,0 +1,21 @@
+int main() {
+  const char *s = "forgreatjustice";
+  unsigned int add = 1;
+  
+  s += add;
+  if (*s != 'o') return 1;
+
+  s += 1UL;
+  if (*s != 'r') return 2;
+
+  const char *s2 = (s += add);
+  if (*s2 != 'g') return 3;
+
+  s2 -= add;
+  if (*s2 != 'r') return 4;
+
+  return 0;
+}
+
+// run-translated-c
+// c_frontend=clang

